app/models/service_template_transformation_plan.rb

```diff
@@ -73,11 +78,26 @@ def self.validate_config_info(options)
 
     raise _('Must provide an existing transformation mapping') if mapping.blank?
 
-    vms = if config_info[:vm_ids]
-            VmOrTemplate.find(config_info[:vm_ids])
-          else
-            config_info[:vms]
-          end
+    pre_service_id  = config_info[:pre_service] ? config_info[:pre_service].id : config_info[:pre_service_id]
+    post_service_id = config_info[:post_service] ? config_info[:post_service_id].id : config_info[:post_service_id]
+
+    vms = []
+    if config_info[:tasks]
+      vm_objects = VmOrTemplate.where(:id => config_info[:tasks].collect { |vm_hash| vm_hash[:vm_id] }.compact)
+      config_info[:tasks].each do |vm_hash|
+        vm_obj = if vm_hash.key?(:vm_id)
+                   vm_objects.find_by(:id => vm_hash[:vm_id])
+                 else
+                   vm_hash[:vm]
+                 end
+        next if vm_obj.nil?
+
+        vm_options = {}
+        vm_options[:pre_ansible_playbook_service_template_id] = pre_service_id if vm_hash[:pre_service]
```

Comment:
@bzwei Instead of passing the service template id down into each task would it make sense to just capture the `:pre_service` and `:post_service` flags for each and at runtime do the lookup from the associated plan. 
 Thoughts?

Reply:
Since we only store an id, it makes no big difference to use id vs boolean, but it saves extra logic to resolve the id at runtime.

Reply:
I was thinking about having to keep the id in-sync across all the records if someone edited the plan and selected a new service.  You would not need to adjust each task and the value is really only looked up once during the run so the overhead at that time is minimal.